core/client/fs/src/main/java/alluxio/client/file/MetadataCachingBaseFileSystem.java

```diff
@@ -78,6 +80,9 @@ public MetadataCachingBaseFileSystem(FileSystemContext context) {
     // asynchronously update access time.
     mAccessTimeUpdater = new ThreadPoolExecutor(0, masterClientThreads, THREAD_KEEPALIVE_SECOND,
         TimeUnit.SECONDS, new SynchronousQueue<>());
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName(MetricKey.CLIENT_META_DATA_CACHE_SIZE.getName()),
+        mMetadataCache::size);
```

Comment:
Looks like guava cache is doing some light calculation inside the size(), will it be better to use cachedguage here? @yuzhu 
what's your thoughts on that?

Reply:
@LuQQiu Good suggestion, but I think `10min` is too long, is there a way to cache the metric for a shorter time?

Reply:
@maobaolong 10 mins interval is just a setting. We can overload `registerCachedGaugeIfAbsent` to pass in a time?